docusaurus/docs/React/custom-code-examples/video-integration-100ms.mdx

```diff
@@ -0,0 +1,881 @@
+    <VideoContext.Provider value={store}>{children}</VideoContext.Provider>
+  );
+};
+```
+
+The last thing to prepare is to make this `VideoContext` easily accessible for all the components where we need it. We create an object for using it like this:
+
+```tsx
+export const useVideoContext = () => useContext(VideoContext);
+```
+
+With that, our basic `VideoContext` is ready to be injected into our component tree. Open up `App.tsx` and inject it inside of the `<Channel>` object so that the code will look like this:
+
+```tsx
+/* ... */
+<Channel>
+  <VideoContextProvider>
+    <Window>
+      <ChannelHeader />
+      <MessageList />
+      <MessageInput />
+    </Window>
+    <Thread />
+  </VideoContextProvider>
+</Channel>
+/* ... */
+```
+
+Make sure the import is added to the top:
+
+```tsx
+import { VideoContextProvider } from './context/VideoContext';
+```
+
+With that we have the object ready for use everywhere we need it. Next up is the real implementation of the functionality around calls.
+
+## 5. Add logic for calls
+
+Before we start with the implementation, let’s go through the process of creating calls for the Stream Chat SDK first.
+
+Every channel has a `data` object where we can freely write additional information inside. We’ll use that to add a key called `callActive` that is set to `true` when a call is active and `false` (or simply not present) otherwise.
+
+In addition to that, we’ll add a `callId` to the `data` object when a call is active so that new participants know where to connect to the call. The SDK allows for the creation of calls and tokens with a simple API call, which makes the entire process much easier.
+
+We’ll subscribe to channel events, specifically the `channel.updated` one to get notified when the state of a call changes and we can act accordingly in our UI.
+
+All of this will happen inside of the `VideoContextProvider` object, so let’s get started.
+
+The first thing we do is subscribe to channel events. We need to add two objects for that. Add the following two lines at the beginning of the `VideoContextProvider` (before the creation of the `connectionState`):
+
+```tsx
+const { channel } = useChannelStateContext();
+const { client } = useChatContext();
+```
+
+Make sure to also have the imports at the top:
+
+```tsx
+import { useChannelStateContext, useChatContext } from 'stream-chat-react';
+```
+
+The good thing when we have the `channel` object available is that we can directly check whether there is an active call going on and set our initial state accordingly. So, let’s change the `connectionState` creation to this:
+
+```tsx
+const [connectionState, setConnectionState] = useState(
+  channel.data?.data?.callActive ? ConnectionState.CallAvailable : ConnectionState.NoCall,
+);
+```
+
+With that, we can add a `useEffect` hook to register for `channel.updated` events. Add the code below the `connectionState` creation:
+
+```tsx
+useEffect(() => {
+  const handleChannelUpdate = (event: any) => {
```

Comment:
```suggestion
  const handleChannelUpdate = (event: Event) => {
```

We have a type that comes from the `stream-chat` packate that defines the event object but the `any` might have been intentional so feel free to ignore this comment. :)